tsconfig.json

```diff
@@ -0,0 +1,19 @@
+{
+  "files": [
+    "./x-element.js"
+  ],
+  "compilerOptions": {
+    "target": "ES2023",
+    "allowJs": true,
+    "checkJs": true,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    // Even though the docs suggest that “declarationMap” is a way to go from
+    //  generated-js back to ts… it also seems to be the case that it will help
+    //  IDEs get to the right _js_ when using types. See
+    //  https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html
+    "declarationMap": true,
```

Comment:
After re-reviewing this. I _think_ it’s actually most correct to emit a source map.

1. If you’re using the JS on it’s own — the source map is inert (since our JS is running as-is).
2. If you’re using the TS declaration file, the source map will hint IDEs to route you back to the original JS.

This was tricky to wrap my head around at first since I typically think about these source mappings as a way to go from the _generated JS_ back to the _original TS_ — but, it seems like you can leverage them to hint IDEs to do the opposite as well.